src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java

```diff
@@ -0,0 +1,62 @@
+package com.boozallen.aissemble;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+import io.kokuwa.maven.helm.UploadMojo;
+import io.kokuwa.maven.helm.pojo.HelmRepository;
+import io.kokuwa.maven.helm.pojo.RepoType;
+
+@Mojo(name = "upload-dependencies")
+public class UploadDependenciesMojo extends UploadMojo {
+
+	@Parameter(property = "url", required = true)
+	private String url;
+
+	@Parameter(property = "id", required = true)
+	private String id;
+
+	@Parameter(property = "type", required = true)
+	private RepoType type;
+
+	@Parameter(defaultValue = "${project}", required = true, readonly = true)
+	private MavenProject project;
+
+	@Override
+	public void execute() throws MojoExecutionException {
+		String projectPath = project.getBasedir().getAbsolutePath();
+		if (!isHelmChart(projectPath)) {
+			return;
+		}
+		uploadRepoStable = createHelmRepository(id, url, type);
+		uploadRepoSnapshot = createHelmRepository(id, url, type);
+		outputDirectory = new File(Paths.get(projectPath, "charts").toString());
+		if (!outputDirectory.exists()) {
+			return;
+		}
+		try {
+			super.execute();
+		} catch (Exception e) {
+			getLog().info(String.format("%s:%s", e.getMessage(), e.getCause()));
+		}
+	}
+
+	private boolean isHelmChart(String path) {
+		File chartFile = new File(Paths.get(path, "Chart.yaml").toString());
```

Comment:
C: This doesn't support modules which house multiple Helm charts.  Why not use `getChartDirectories`?

Reply:
Added functionality to incorporate subcharts. Though the isHelmChart method is still needed to not re-deploy previously deployed chart dependencies.